Alerting: [Docs] Clarifications regarding what's supported in alerting message templates (#43544)
grafana/grafana: docs/sources/alerting/unified-alerting/message-templating/_index.md

### Using templates

> The following example shows the use of default templates to render an alert message in Slack. The message title contains a count of firing or resolved alerts and the message body has a list of alerts with status.

Author: This is a lot better. Applied nearly verbatim: "alerts that are firing or were alerts." -> "alerts that are firing or were resolved."

Use caution when deleting a template since Grafana does not prevent you from deleting templates that are in use.

> ### Nested templates

Author: `Embedded templates` maybe? Go's template language docs call this ["Nested templates"](https://pkg.go.dev/text/template#hdr-Nested_template_definitions) so that's what I went with here.

You can embed templates within other templates.

> For example, you can define a template fragment can using the `define` keyword:

Reviewer: ```suggestion
For example, you can define a template fragment using the `define` keyword:
```

```
{{ define "mytemplate" }}
  {{ len .Alerts.Firing }} firing. {{ len .Alerts.Resolved }} resolved.
{{ end }}
```

Other custom templates may then embed the fragment using the `template` keyword:

```
Alert summary:
{{ template "mytemplate" . }}
```

> Grafana provides a few built-in templates, which can be embedded in custom templates:

Reviewer 2: ```suggestion
You can use any of the following built-in template options to embed custom templates.
```

| Name                    | Notes                                                                 |
| ----------------------- | --------------------------------------------------------------------- |
| `default.title`         | Displays high-level status information.                               |
| `default.message`       | Provides a formatted summary of firing and resolved alerts.           |

> | `teams.default.message` | Similar content to `default.messsage`, formatted for Microsoft Teams. |

Reviewer 2: ```suggestion
| `teams.default.message` |Similar to `default.messsage`, formatted for Microsoft Teams. |
```

### Custom template examples

> Below are some sample usages of custom templates.

Reviewer 2: ```suggestion
Here are a few examples of how to use custom templates.
```

Template to render entire notification message:

```
{{ define "mymessage" }}
  {{ if gt (len .Alerts.Firing) 0 }}
    {{ len .Alerts.Firing }} firing:
    {{ range .Alerts.Firing }} {{ template "myalert" .}} {{ end }}
  {{ end }}
  {{ if gt (len .Alerts.Resolved) 0 }}
    {{ len .Alerts.Resolved }} resolved:
    {{ range .Alerts.Resolved }} {{ template "myalert" .}} {{ end }}
  {{ end }}
{{ end }}
```

### HTML in Message Templates

> HTML in Alerting message templates is escaped, and rendering of HTML in the resulting notification is not supported.

Reviewer 2: ```suggestion
Although HTML in alerting message templates is escaped, we do not support rendering of HTML in the resulting notification.
```

Author: I considered dropping the "Although", what are your thoughts on this? The fact that HTML is escaped is why we don't support rendering of HTML, it isn't really an alternative.

> Some notifiers which support rendered HTML (such as the body of emails) support alternative methods of changing the look and feel of the resulting notification. For example, the base template used in Alerting emails is installed to `/usr/share/grafana/public/emails` on Linux.

Reviewer 2: ```suggestion
Some notifiers that support rendered HTML (in the body of emails) support alternative methods of changing the look and feel of the resulting notification. For example, it may install the base template for alerting emails to `/usr/share/grafana/public/emails` on Linux.
```

Author: On second read, my wording in this whole block isn't the greatest. I only mentioned one OS, totally forgot the filename etc. How about:

Some notifiers support alternative methods of changing the look and feel of the resulting notification. For example, Grafana installs the base template for alerting emails to `<grafana-install-dir>/public/emails/ng_alert_notification.html`.
